### PR TITLE
Refactored FT 1.0 Bulkhead Tests with CountDownLatch

### PR DESCRIPTION
**Pull request by scottcurtis2605**
Refactored BulkheadTest.java methods, testAsyncBulkhead and testAsyncBulkheadQueueFull
to utilize CountDownLatch objects. These ensure that tasks within this test
do not always take 10000ms to run in the case of early completion,
or that tests exceed 10000ms in the case of interrupted execution.

AsyncTestFunction.java was also refactored with a new constructor to
enable the creation of an AsyncTestFunction with three CountDownLatch
objects. This enables the new behaviour added to BulkheadTest.java.

### CODE REVIEW

**Review by Azquelt — CHANGES_REQUESTED** [review]
This looks really good, code looks correct and your comments are on point.

However, you need to update the date in the copyright header at the top of both files you've changed. The copyright header needs to contain the year the file was created and the year it was last updated.

File: dev/com.ibm.ws.microprofile.faulttolerance.1.0/test/src/com/ibm/ws/microprofile/faulttolerance/test/BulkheadTest.java [hunk 1]
```diff
@@ -249,28 +268,47 @@ public void testAsyncBulkheadQueueFull() throws InterruptedException, ExecutionE
...
+            AsyncTestFunction callable = new AsyncTestFunction(Duration.ofMillis(10000), isRunningLatch, mayCompleteLatch, completedLatch, id);
+
+            // Final task should be rejected from queue, and throw a Bulkhead exception
             try {
                 futures[4] = executor.execute(callable, context);
                 System.out.println(System.currentTimeMillis() + " Test " + id + " - submitted");
                 fail("Exception not thrown");
             } catch (BulkheadException e) {
-                //expected
+                assertTrue("Should fail with bulkhead exception ", e instanceof BulkheadException);
```
[the remark this answers is not in the document]
Comment (scottcurtis2605) [reply dev/com.ibm.ws.microprofile.faulttolerance.1.0/test/src/com/ibm/ws/microprofile/faulttolerance/test/BulkheadTest.java:306]: Thank you for noting this, I'll get around to making the changes now!

[the remark this answers is not in the document]
Comment (scottcurtis2605) [reply dev/com.ibm.ws.microprofile.faulttolerance.1.0/test/src/com/ibm/ws/microprofile/faulttolerance/test/BulkheadTest.java]: Changes have just been made to update the copyright header, and remove the assertion.

**Review by Azquelt — APPROVED** [review]

**Comment by scottcurtis2605** [comment]
#build

**Comment by scottcurtis2605** [comment]
#run-libby-bot

**Pull request merged**